### ISSUE TITLE
fast-button elements on the same line with and without icons do not align

### ISSUE THREAD

**Issue by rajsite** [microsoft/fast#5414]
# 🐛 Bug Report

See the following screenshot of two FAST Buttons on the same line (one with icon and one without) and how they are offset from each other:
![image](https://user-images.githubusercontent.com/1588923/142709089-197bad98-f605-4b34-8175-614ac991bce0.png)

## 💻 Repro or Code Sample

Stackblitz: https://stackblitz.com/edit/fast-button-alignment?file=index.html

```html
<div>
  <fast-button>Hello</fast-button>
  <fast-button>
    World
    <svg slot="start" width="16" height="16" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg"><!-- ... --></svg>
  </fast-button>
</div>
```

## 🤔 Expected Behavior

I expect the buttons to be aligned on the same line like the following workaround:

```html
<div>
  <fast-button style="vertical-align:middle;">Hello</fast-button>
  <fast-button style="vertical-align:middle;">
    World
    <svg slot="start" width="16" height="16" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg"><!-- ... --></svg>
  </fast-button>
</div>
```

Note:  See the added `style="vertical-align:middle;"`

![image](https://user-images.githubusercontent.com/1588923/142709219-c31d50cb-809f-4b1f-93f2-afbfc7c54a68.png)

## 😯 Current Behavior

See above.

## 💁 Possible Solution

See workaround above.

## 🔦 Context

We style buttons based on fast-foundation and ran into a similar issue when using [display: inline-flex](https://github.com/microsoft/fast/blob/c60e278fafac6e9d9be2e634a8537097dae81e96/packages/web-components/fast-components/src/styles/patterns/button.ts#L42) like fast is doing and have included  a `:host {vertical-align: middle; }` in our styles. I think it has to do with weirdness involving changing the display configuration of the inner `<button>` but we did not find a clear root cause.

## 🌍 Your Environment

    "@microsoft/fast-components": "^2.14.2",
    "@microsoft/fast-element": "^1.6.1"

**Comment by alphashr** [comment]
I think I know the fix. 
Following changes would align the contents (all three slots: start, content and end) on cross axis properly: 
```css
.control {
    align-items: baseline
}

.content,
.start,
.end {
    align-self: center;
}
```
![image](https://user-images.githubusercontent.com/54422255/142711564-add8fa66-05e0-4abe-b2e8-ec61075539ae.png)


I'd be happy to pick it up once it's triaged.

**Comment by alphashr** [comment]
I can see this issue for ```search``` component (not yet released) which uses the same startSlotTemplate and endSlotTemplate like button:
![image](https://user-images.githubusercontent.com/54422255/142712650-6b4151f9-24d8-44aa-8baf-d9b4becdfbe4.png)
First control:
```html
<fast-search minlength="8"></fast-search>
```
Second one with svg in its start slot:
```html
<fast-search>
    <svg slot="start" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
        <path
            d="M6.5 7.7h-1v-1h1v1zm4.1 0h-1v-1h1v1zm4.1-1v2.1h-1v2.6l-.1.6-.3.5c-.1.1-.3.3-.5.3l-.6.1H10l-3.5 3v-3H3.9l-.6-.1-.5-.3c-.1-.1-.3-.3-.3-.5l-.1-.6V8.8h-1V6.7h1V5.2l.1-.6.3-.5c.1-.1.3-.3.5-.3l.6-.1h3.6V1.9a.8.8 0 01-.4-.4L7 1V.6l.2-.3.3-.2L8 0l.4.1.3.2.3.3V1l-.1.5-.4.4v1.7h3.6l.6.1.5.3c.1.1.3.3.3.5l.1.6v1.5h1.1zm-2.1-1.5l-.2-.4-.4-.2H3.9l-.4.2-.1.4v6.2l.2.4.4.2h3.6v1.8L9.7 12h2.5l.4-.2.2-.4V5.2zM5.8 8.9l1 .7 1.2.2a5 5 0 001.2-.2l1-.7.7.7c-.4.4-.8.7-1.4.9-.5.2-1 .3-1.6.3s-1.1-.1-1.6-.3A3 3 0 015 9.6l.8-.7z"
        />
    </svg>
</fast-search>
```
Other components such as ```Breadcrumb```, ```Listbox Option```, ```Number Field```, ```Text Field``` and maybe other components have the same issue.

**Comment by chrisdholt** [comment]
I believe this was closed by #5435 - @rajsite if you come across other instances, let us know, but this should be addressed. Thanks all!
